jcl/src/jdk.jcmd/share/classes/openj9/tools/attach/diagnostics/tools/Jstack.java

```diff
@@ -0,0 +1,157 @@
+	/**
+	 * Print a list of Java processes and information about them.
+	 * @param args Arguments to the application
+	 */
+	public static void main(String[] args) {
+		PrintStream out = System.out;
+
+		if (!parseArguments(args)) {
+			System.exit(1);
+		}
+		AttacherDiagnosticsProvider diagProvider = new AttacherDiagnosticsProvider();
+
+		String myId = AttachHandler.getVmId();
+		out.println(LocalDateTime.now());
+		for (String vmid: vmids) {
+			if (vmid.equals(myId)) {
+				continue;
+			}
+			/* if the ID looks like a process ID, check if it is running */
+			if (vmid.matches("\\d+")) { //$NON-NLS-1$
+				long pid = Long.parseLong(vmid);
+				if (!IPC.processExists(pid)) {
+					continue;
+				}
+			}
+			try {
+				diagProvider.attach(vmid);
+				DiagnosticsInfo groupInfo = diagProvider.getThreadGroupInfo(printSynchronizers);
+				out.printf("Virtual machine: %s JVM information %s%n", vmid, groupInfo.getJavaInfo()); //$NON-NLS-1$
+				out.println(groupInfo.toString());
+				if (printProperties) {
+					out.println("System properties:"); //$NON-NLS-1$
+					out.println(diagProvider.getSystemProperties());
+					out.println("Agent properties:"); //$NON-NLS-1$
+					out.println(diagProvider.getAgentProperties());
+				}
+			} catch (Exception e) {
+				System.err.printf("Error getting data from %s", vmid); //$NON-NLS-1$
+				final String msg = e.getMessage();
+				if (null == msg) {
+					System.err.println();
+				} else {
+					if (msg.matches(IPC.INCOMPATIBLE_JAVA_VERSION)) {
+						System.err.println(": incompatible target JVM"); //$NON-NLS-1$
+					} else {
+						System.err.printf(": %s%n", msg); //$NON-NLS-1$
+					}
+				}
+				if (DiagnosticProperties.isDebug) {
+					e.printStackTrace();
+				}
+			} finally {
+				try {
+					diagProvider.detach();
+				} catch (IOException e) {
+					/* ignore */
+				}
+			}
+		}
+	}
+
+	@SuppressWarnings("nls")
+	private static boolean parseArguments(String[] args) {
+		boolean okay = true;
+		printProperties = DiagnosticProperties.isDebug;
+		printSynchronizers = false;
+		final String HELPTEXT = "jstack: listing thread information about another Java process%n"
+				+ " Usage:%n"
+				+ "    jstack <vmid>*%n"
+				+ "        <vmid>: Attach API VM ID as shown in jps or other Attach API-based tools%n"
+				+ "        <vmid>s are read from stdin if none are supplied as arguments%n"
+				+ "    -p: print the target's system and agent properties%n"
+				+ "    -l: Long format. Print the thread's ownable synchronizers%n"
+				+ "    -J: supply arguments to the Java VM running jps%n"
+				;
+		vmids = new ArrayList<>();
+		for (String a: args) {
+			if (!a.startsWith("-")) {
+				vmids.add(a);
+			} else if (a.equals("-p")) {
+				printProperties = true;
+			} else if (a.equals("-l")) {
+				printSynchronizers = true;
+			} else {
+				System.err.printf(HELPTEXT);
+				okay = false;
+				break;
+			}
+		}
+
+		if (okay && vmids.isEmpty()) {
+			/* grab the VMIDs from stdin. Omit my own ID. */
```

Comment:
The code below does *not* omit the executor's id. Please fix the code or the comment.